--- jquery.flexbox.js.orig	2010-11-24 13:03:02.000000000 +0900
+++ jquery.flexbox.1.5.js	2010-12-29 01:04:23.000000000 +0900
@@ -1,3 +1,8 @@
+/*
+ * This file contains some modification written by Yasuhiro ABE <yasu@yasundial.org>
+ * Copyright (c) 2010 Yasuhiro ABE, http://www.yadiary.net/
+ * Original copyright is following;
+ */
 /*!
 * jQuery FlexBox $Version: 0.9.6 $
 *
@@ -53,6 +58,8 @@
         scrolling = false,
         pageSize = o.paging && o.paging.pageSize ? o.paging.pageSize : 0,
 		retrievingRemoteData = false,
+	queryKey = o.queryKey,   // first key string against the query, added Yasuhiro ABE
+	queryKey2 = o.queryKey2, // second key string, added Yasuhiro ABE
         $div = $(div).css('position', 'relative').css('z-index', 0);
 
         // The hiddenField MUST be appended to the div before the input, or IE7 does not shift the dropdown below the input field (it overlaps)
@@ -244,15 +251,20 @@
 				if ($content.outerHeight() > 0)
 					$content.css('height', $content.outerHeight());
                 $content.html('').attr('scrollTop', 0);
-				
-                var cached = checkCache(q, p);
+		var k1 = queryKey && $(queryKey).val() ? $(queryKey).val() : '';
+		var k2 = queryKey2 && $(queryKey2).val() ? $(queryKey2).val() : '';
+		var kk = k2 === '' ? k1 : k2;
+                var cached = checkCache(kk + delim + q, p);
                 if (cached) {
 					$content.css('height', 'auto');
                     displayItems(cached.data, q);
                     showPaging(p, cached.t);
                 }
                 else {
-                    var params = { q: q, p: p, s: pageSize, contentType: 'application/json; charset=utf-8' };
+		    var k1 = queryKey && $(queryKey).val() ? $(queryKey).val() : '';
+		    var k2 = queryKey2 && $(queryKey2).val() ? $(queryKey2).val() : '';
+		    var kk = k2 === '' ? k1 : k2;
+		    var params = { q: q, p: p, s: pageSize, contentType: 'application/json; charset=utf-8', k: k1, j: k2 };
                     var callback = function(data, overrideQuery) {
                         if (overrideQuery === true) q = overrideQuery; // must compare to boolean because by default, the string value "success" is passed when the jQuery $.getJSON method's callback is called
                         var totalResults = parseInt(data[o.totalProperty]);
@@ -270,12 +282,12 @@
                                 pageData[o.totalProperty] = totalResults;
                                 pageData[o.resultsProperty] = data[o.resultsProperty].splice(0, pageSize);
                                 if (i === 1) totalSize = displayItems(pageData, q);
-                                updateCache(q, i, pageSize, totalResults, pageData, totalSize);
+                                updateCache(kk + delim + q, i, pageSize, totalResults, pageData, totalSize);
                             }
                         }
                         else {
                             var totalSize = displayItems(data, q);
-                            updateCache(q, p, pageSize, totalResults, data, totalSize);
+                            updateCache(kk + delim + q, p, pageSize, totalResults, data, totalSize);
                         }
                         showPaging(p, totalResults);
 						$content.css('height', 'auto');
@@ -848,7 +860,9 @@
             showSummary: true, // whether to show 'displaying 1-10 of 200 results' text
             summaryClass: 'summary', // class for 'displaying 1-10 of 200 results', prefix with containerClass
             summaryTemplate: 'Displaying {start}-{end} of {total} results' // can use {page} and {pages} as well
-        }
+        },
+	queryKey: '', // used for additional parameter of the query parameter, added by Yasuhiro ABE
+	queryKey2: '' // used for additional parameter of the query parameter, added by Yasuhiro ABE
     };
 
     $.fn.setValue = function(val) {
